### PR TITLE
Set-up TypeScript codegen and publishing

### LINKED ISSUE #61: Do a JavaScript release
See #3.

This would entail:

- setting up an npm project for this repo
- automating the procedure to generate JS bindings to the protos (`Makefile`), generally checking in the generated code (`gen/` and `.github/workflows/`)
- automating/documenting the procedure to do releases of the generated code (`RELEASE.md` plus any GH Actions) on a tag, preferably `releases/js/v0.1.1` or whatever.

Then you could start consuming it in sigstore-js.

 CC @bdehamer @feelepxyz @eddiezane

### PR DESCRIPTION
**Pull request by bdehamer**
#### Summary
Set-up code generation and publishing for TypeScript types.

The npm package will be published to https://www.npmjs.com/package/@sigstore/protobuf-specs

**NOTE:** Before we can publish the npm package I'll need someone to set-up a repository secret for the `NPM_PUBLISH_TOKEN` value.

Closes #61

### CODE REVIEW

File: .github/workflows/typescript-build.yml
Comment (znewman01) [inline .github/workflows/typescript-build.yml:42]: Do we need to run on all platforms? IMO just `ubuntu-latest` would be ok since we're *really* just checking that the release works IIUC.

File: gen/pb-typescript/README.md
Comment (znewman01) [inline gen/pb-typescript/README.md:3]: nit: `protbuf`

File: gen/pb-typescript/LICENSE
Comment (znewman01) [inline gen/pb-typescript/LICENSE:2]: Why is this needed?

**Review by znewman01 — APPROVED** [review]
LGTM, couple minor questions.

Once this is merged please coordinate with one of the owners to get the secret added: https://github.com/orgs/sigstore/teams/protobuf-specs-codeowners

(I'm OOO next week)

File: .github/workflows/typescript-build.yml
Comment (bdehamer) [reply .github/workflows/typescript-build.yml]: Yeah, good catch. I'd just lifted this out of the `sigstore-js` matrix without giving it much thought. Will pare it down.

File: gen/pb-typescript/README.md
Comment (bdehamer) [reply gen/pb-typescript/README.md]: Fixed

File: gen/pb-typescript/LICENSE
Comment (bdehamer) [reply gen/pb-typescript/LICENSE:2]: This ensures that the `LICENSE` gets included in the npm package.

**Review by znewman01 — APPROVED** [review]

**Pull request merged**
